common/src/node/ethereum/utils.rs

```diff
@@ -321,9 +360,21 @@ pub async fn broadcast_contract_transfer_tx(
         } => {
             let to_address = address_from_str(&to_address)?;
             let amount = u256_from_dec_str(&amount)?;
-            let contract = Contract::new_erc20(&contract_address, client)?;
-            let call = contract.transfer(to_address, amount);
-            ContractCall::from(call).legacy(legacy).send().await
+            let client = get_ethers_provider(web3api_url).await?;
+            if let Some(key) = secret_key {
+                let provider = client.interval(Duration::from_millis(polling_interval_ms));
+                let wallet = LocalWallet::from(key.get_signing_key()).with_chain_id(chain_id);
+                let client = SignerMiddleware::new(provider, wallet);
```

Comment:
these three liens are duplicated in all of the functions, maybe it can be put in a private helper to create the client?

Reply:
ok, i'll modify